src/mode.js

```diff
@@ -89,15 +89,7 @@ function getMode_(win) {
   // paths for localhost/testing/development are eliminated.
   return {
     localDev: isLocalDev,
-    // Triggers validation or enable pub level logging. Validation can be
-    // bypassed via #validate=0.
-    // Note that AMP_DEV_MODE flag is used for testing purposes.
-    // Use Array.indexOf instead of Array.includes because of #24219
-    development: !!(
-      ['1', 'actions', 'amp', 'amp4ads', 'amp4email'].indexOf(
-        hashQuery['development']
-      ) >= 0 || win.AMP_DEV_MODE
-    ),
+    development: isModeDevelopment(win),
```

Comment:
I wonder if we can get rid of this entirely? Or move it into the binaries that need it.

Reply:
The plan is to turn all the mode properties into DCE-able helpers that can be consumed directly and only included in binaries that use it.

Reply:
@jridgewell @rcebulko I didn't plan to change all the `getMode().development` calls into `getModeDeveloppment()` throughout the codebase in this PR, is that sth that we'd want to do now?

Reply:
Nope, it's good that most of the codebase accesses via `getMode()`. The dependencies required for that don't exist in `core` yet :)